File: backend/app/schemas/reports.py
from datetime import date, datetime

from pydantic import BaseModel

CORE_METRICS: tuple[str, ...] = (
    "Revenue",
    "Room Revenue",
    "Occupancy",
    "ADR",
    "RevPAR",
)


class ReportInventoryItem(BaseModel):
    report_id: int
    report_date: date
    file_name: str
    uploaded_at: datetime
    hotel_count: int
    metrics_present: list[str]
    metric_row_counts: dict[str, int]
    parse_status: str  # full | partial | revenue_only | empty
    parse_label: str
    is_complete: bool


class ReportInventoryResponse(BaseModel):
    total: int
    full_parse_count: int
    reports: list[ReportInventoryItem]

File: frontend/src/lib/reports.ts
import { apiFetch } from "@/lib/auth";

export type ReportInventoryItem = {
  report_id: number;
  report_date: string;
  file_name: string;
  uploaded_at: string;
  hotel_count: number;
  metrics_present: string[];
  metric_row_counts: Record<string, number>;
  parse_status: "full" | "partial" | "revenue_only" | "empty";
  parse_label: string;
  is_complete: boolean;
};

export type ReportInventoryResponse = {
  total: number;
  full_parse_count: number;
  reports: ReportInventoryItem[];
};

export async function fetchReportsInventory(): Promise<ReportInventoryResponse> {
  const response = await apiFetch("/reports/inventory", { cache: "no-store" });
  if (!response.ok) {
    throw new Error("Failed to load reports inventory");
  }
  return response.json();
}

export function formatUploadedAt(iso: string): string {
  const d = new Date(iso);
  return d.toLocaleString("en-US", {
    month: "short",
    day: "numeric",
    year: "numeric",
    hour: "numeric",
    minute: "2-digit",
  });
}
